cmd/siac/main.go

```diff
@@ -309,8 +134,15 @@ func main() {
 	root.AddCommand(bashcomplCmd)
 	root.AddCommand(mangenCmd)
 
-	// parse flags
-	root.PersistentFlags().StringVarP(&addr, "addr", "a", "localhost:9980", "which host/port to communicate with (i.e. the host/port siad is listening on)")
+	// Check if the api password environment variable is set.
+	apiPassword := os.Getenv("SIA_API_PASSWORD")
+	if apiPassword != "" {
+		fmt.Println("Using SIA_API_PASSWORD environment variable")
+	}
+	// initialize client
+	root.PersistentFlags().StringVarP(&httpClient.Address, "addr", "a", "localhost:9980", "which host/port to communicate with (i.e. the host/port siad is listening on)")
+	root.PersistentFlags().StringVarP(&httpClient.Password, "apipassword", "x", apiPassword, "the password for the API's http authentication")
+	root.PersistentFlags().StringVarP(&httpClient.UserAgent, "useragent", "u", "Sia-Agent", "the useragent used by siac to connect to the daemon's API")
```

Comment:
`x` isn't a great shorthand. Also, `u` is used for BasicAuth in `curl`, so that couuld be confusing.

Shorthands are optional; maybe we shouldn't have shorthands for these flags.